exported_locals.go

```diff
@@ -187,13 +188,15 @@ func (r exportedLocalExprs) eval(meta StackMetadata, globals *Globals) (Exported
 
 	logger.Trace().
 		Msg("Add proper namespace for stack metadata evaluation.")
-	if err := meta.SetOnEvalCtx(evalctx); err != nil {
+	err := evalctx.SetNamespace("terramate", meta.ToCtyMap())
```

Comment:
I think this will also be duplicated on globals eval now (why I had the SetOnEvalCtx, which I didn't liked anyway :laughing: ). Maybe we could have our own evalctx instance that is more domain specific and has something like:

```
evalctx.SetMetadata(meta)
evalctx.SetGlobals(globals)
```

And the namespace names + how to translate from meta/globals to hcl eval stays there. It could do the ToCtyMap() for stack.Metadata, so it would not need any Cty specific knowledge. This was not well handled before either...just thinking if now we can do something better (I don't like spreading these cty stuff around x_x..but didn't had a good idea for that).

Reply:
yeah we can do this in a separate PR, not sure yet where this guy fits.

Reply:
man..me neither... I was honestly annoyed...like..where the fuck does this fit ...no good idea yet :laughing: